python/sanlock.c

```diff
@@ -395,14 +436,15 @@ py_init_resource(PyObject *self __unused, PyObject *args, PyObject *keywds)
 
     if (rv != 0) {
         __set_exception(rv, "Sanlock resource init failure");
-        goto exit_fail;
+        goto finally;
     }
 
-    free(res);
-    Py_RETURN_NONE;
-
-exit_fail:
-    free(res);
+finally:
+    Py_XDECREF(lockspace);
+    Py_XDECREF(resource);
+    if (res) free(res);
```

Comment:
free(NULL) is ok, we don't need the if, but looks like this is the way free() is used in this module, so we can improve this later.